app/controllers/comments_controller.rb

```diff
@@ -1,10 +1,10 @@
 class CommentsController < ApplicationController
   def create
     @comment = current_user.comments.new(comment_params)
-    @post = Post.find(params[:comment][:post_id])
+    @post = Post.find(comment_params[:post_id])
 
     if @comment.save
-      redirect_to "#{users_path}/#{@post.author_id}/posts/#{@post.id}"
+      redirect_to user_post_path @post
     else
       render :new, status: :unprocessable_entity
```

Comment:
Please solve the `N+1` problem when fetching all posts and their comments for a user in the `posts#index` action.
For example:
You may use `@posts = @user.posts.includes(:comments)` instead of `@posts = @user.posts` in `posts#index` action.
The most common solution to this problem is eager loading. You may learn more about it [here](https://medium.com/idopterlabs/solving-the-n-1-problem-in-rails-b5ea677e78e0).